pkg/distributor/distributor.go

```diff
@@ -1378,25 +1341,52 @@ func (d *Distributor) push(ctx context.Context, pushReq *Request) error {
 	return err
 }
 
-func preallocSliceIfNeeded[T any](size int) []T {
-	if size > 0 {
-		return make([]T, 0, size)
-	}
-	return nil
+// getSeriesAndMetadataTokensForBatchRequest returns a slice of tokens for the series and metadata from the request in this specific order.
+// Metadata tokens start at initialMetadataIndex.
+func getSeriesAndMetadataTokensForBatchRequest(userID string, req *mimirpb.WriteRequest) (keys []uint32, initialMetadataIndex int) {
```

Comment:
What makes it a "BatchRequest" ?  This is called with the entire request; I thought the batches were what DoBatch splits it into.

Reply:
My reasoning was: _since we pass this to ring.DoBatch_ ... but if this raised you a question, then it's not helping. I'm going to remove the `ForBatchRequest` suffix from this method.